org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/capabilities/XMLCapabilityManager.java

```diff
@@ -153,9 +159,20 @@ public void initializeCapabilities() {
 		if (this.getClientCapabilities().isReferencesDynamicRegistrationSupported()) {
 			registerCapability(REFERENCES_ID, TEXT_DOCUMENT_REFERENCES);
 		}
+		if (this.getClientCapabilities().isDidChangeWatchedFilesRegistered()) {
+			registerWatchedFiles();
+		}
+		
 		syncDynamicCapabilitiesWithPreferences();
 	}
 
+	private void registerWatchedFiles() {
+		List<FileSystemWatcher> watchers = new ArrayList<>();
```

Comment:
`new ArrayList<>(1);` if you know the list size in advance,

If you now the size will be 1, then you can also do:
```java
List<FileSystemWatcher> watchers = Collections.singletonList(new FileSystemWatcher("**/*.xsd")));
```